src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -27,20 +31,64 @@ internal DeflateStream(Stream stream, CompressionMode mode, long uncompressedSiz
+        /// <exception cref="System.ArgumentException"><paramref name="mode" /> is not a valid <see cref="System.IO.Compression.CompressionMode" /> value.
+        /// -or-
+        /// <see cref="System.IO.Compression.CompressionMode" /> is <see cref="System.IO.Compression.CompressionMode.Compress" /> and <see cref="System.IO.Stream.CanWrite" /> is <see langword="false" />.
+        /// -or-
+        /// <see cref="System.IO.Compression.CompressionMode" /> is <see cref="System.IO.Compression.CompressionMode.Decompress" /> and <see cref="System.IO.Stream.CanRead" /> is <see langword="false" />.</exception>
         public DeflateStream(Stream stream, CompressionMode mode, bool leaveOpen) : this(stream, mode, leaveOpen, ZLibNative.Deflate_DefaultWindowBits)
         {
         }
 
-        // Implies mode = Compress
+        /// <summary>Initializes a new instance of the <see cref="System.IO.Compression.DeflateStream" /> class by using the specified stream and compression level.</summary>
+        /// <param name="stream">The stream to compress.</param>
+        /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
+        /// <remarks>You use this constructor when you want to specify whether compression efficiency or speed is more important for an instance of the <see cref="System.IO.Compression.DeflateStream" /> class.
+        /// This constructor overload uses the compression mode <see cref="System.IO.Compression.CompressionMode.Compress" />. To set the compression mode to another value, use the <see cref="System.IO.Compression.DeflateStream.DeflateStream(System.IO.Stream,System.IO.Compression.CompressionMode)" /> or <see cref="System.IO.Compression.DeflateStream.DeflateStream(System.IO.Stream,System.IO.Compression.CompressionMode,bool)" /> overload.</remarks>
+        /// <example>The following example shows how to set the compression level when creating a <see cref="System.IO.Compression.DeflateStream" /> object.
+        /// <format type="text/markdown"><![CDATA[
+        /// [!code-csharp[System.IO.Compression.DeflateStream#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.io.compression.deflatestream/cs/program1.cs#1)]
+        /// [!code-vb[System.IO.Compression.DeflateStream#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.deflatestream/vb/program1.vb#1)]
+        /// ]]></format></example>
+        /// <exception cref="System.ArgumentNullException"><paramref name="stream" /> is <see langword="null" />.</exception>
+        /// <exception cref="System.ArgumentException">The stream does not support write operations such as compression. (The <see cref="System.IO.Stream.CanWrite" /> property on the stream object is <see langword="false" />.)</exception>
         public DeflateStream(Stream stream, CompressionLevel compressionLevel) : this(stream, compressionLevel, leaveOpen: false)
```

Comment:
Double slash comments should probably stay, but I'll move them here:
```suggestion
        // Implies mode = Compress
        public DeflateStream(Stream stream, CompressionLevel compressionLevel) : this(stream, compressionLevel, leaveOpen: false)
```